### PR TITLE
Revert broadcast optimization to prepare for p2p NET

### PR DESCRIPTION
**Pull request by nieznanysprawiciel**
Don't merge until the final decision, because we must still care a little bit about central NET behavior

resolves: https://github.com/golemfactory/ya-relay/issues/15

### CODE REVIEW

**Review by maaktweluit — APPROVED** [review]
LGTM!

File: core/market/src/protocol/discovery.rs [hunk 1]
```diff
@@ -228,10 +228,21 @@ impl Discovery {
             },
         );
         // Subscribe to receiving broadcasts only when demand is subscribed for the first time.
```
Comment (jiivan) [inline core/market/src/protocol/discovery.rs:230]:
```suggestion
        // Subscribe to offers broadcast.
```

File: core/market/src/protocol/discovery.rs [hunk 2]
```diff
@@ -228,10 +228,21 @@ impl Discovery {
...
-        };
+        {
+            let mut prefix_guard = self.inner.lazy_binder_prefix.lock().await;
+            if let Some(old_prefix) = (*prefix_guard).replace(local_prefix.to_string()) {
+                log::info!("Dropping previous lazy_binder_prefix, and replacing it with new one. old={}, new={}", old_prefix, local_prefix);
+            };
+        }
+
+        // We don't lazy bind broadcasts handlers anymore on first Demand creation.
+        // But we still have option to do this easy in the future.
```
Comment (jiivan) [inline core/market/src/protocol/discovery.rs:239]:
```suggestion
        // But we still have option to do this easily in the future.
```

File: core/market/src/protocol/discovery.rs [hunk 3]
```diff
@@ -228,10 +228,21 @@ impl Discovery {
...
+        self.lazy_bind_gsb().await.map_or_else(
```
Comment (jiivan) [inline core/market/src/protocol/discovery.rs:240]:
```suggestion
        self.bind_gsb_offers_broadcast().await.map_or_else(
```

**Review by jiivan — DISMISSED** [review]
LGTM! Altough I would rename the lazy broadcast method as it's not lazy anymore - just broadcast subscription (and change the trace log from inside this function to not mention laziness).

File: core/market/src/protocol/discovery.rs [hunk 4]
```diff
@@ -247,7 +247,7 @@ impl Discovery {
         Ok(())
     }
 
-    pub async fn lazy_bind_gsb(&self) -> Result<(), DiscoveryInitError> {
+    pub async fn bind_gsb_broadcast(&self) -> Result<(), DiscoveryInitError> {
         log::trace!("LazyBroadcastBind");
```
Comment (jiivan) [inline core/market/src/protocol/discovery.rs:251]:
```suggestion
        log::trace!("GsbBroadcastBind");
```

**Review by jiivan — APPROVED** [review]

**Pull request merged**